File: ROS_YuXiang/test2.4_workspace/src/demo_python_pkg/demo_python_pkg/person_node.py
class PersonNode:
    def __init__(self, name: str, age: int) -> None:
        print(f"{name} is {age} years old")
        self.name = name
        self.age = age

    def eat(self, food: str) -> None:
        print(f"{self.name} is eating {food}")

def main():
    node = PersonNode("Alice", 18)
    node.eat("apple")
        

File: ROS_YuXiang/test2.4_workspace/src/demo_cpp_pkg/src/person_node.cpp
#include <string>
#include "rclcpp/rclcpp.hpp"

class PersonNode : public rclcpp::Node
{
private:
    std::string name_;
    int age_;

public:
    PersonNode(const std::string &node_name,
               const std::string &name,
               const int &age) : Node(node_name)
    {
        this->name_ = name;
        this->age_ = age;
    };

    void eat(const std::string &food)
    {
        RCLCPP_INFO(this->get_logger(), "I'm %s, %d years old, I'm eating %s.",
                    this->name_.c_str(), this->age_, food.c_str());
    };
};


int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("person_cpp_node", "Tom", 18);
    node->eat("apple");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}